src/frontend/src/instance/distributed.rs

```diff
@@ -503,4 +576,106 @@ ENGINE=mito",
             }
         }
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_show_databases() {
+        let (dist_instance, _) = create_dist_instance().await;
+
+        let sql = "create database test_show_databases";
+        let output = dist_instance
+            .handle_sql(sql, QueryContext::new_ref())
+            .await
+            .unwrap();
+        match output {
+            Output::AffectedRows(rows) => assert_eq!(rows, 1),
+            _ => unreachable!(),
+        }
+
+        let sql = "show databases";
+        let output = dist_instance
+            .handle_sql(sql, QueryContext::new_ref())
+            .await
+            .unwrap();
+        match output {
+            Output::RecordBatches(r) => {
+                let expected1 = vec![
+                    "+---------------------+",
+                    "| Schemas             |",
+                    "+---------------------+",
+                    "| public              |",
+                    "| test_show_databases |",
+                    "+---------------------+",
+                ];
+                let expected2 = vec![
+                    "+---------------------+",
+                    "| Schemas             |",
+                    "+---------------------+",
+                    "| test_show_databases |",
+                    "| public              |",
+                    "+---------------------+",
+                ];
+                let pretty = r.pretty_print();
+                let lines = pretty.lines().collect::<Vec<_>>();
+                assert!(lines == expected1 || lines == expected2)
```

Comment:
These rendered strings are difficult to maintain, please try to avoid to do assertion using this approach.

Reply:
I think this "rendered strings" are better because it's far more simple and straightforward for us human to read the testing codes, and verifying the assertions' correction. Otherwise, people have to read the disassembled `RecordBatch` in testing codes and "assemble" them in their minds, which adds another indirection to understand.

That said, testing in this way does have its shortcomings in that it can't verify the datatypes of the output. But I think it's tolerable, because after all, we are presenting the output in text protocol in almost all protocols to clients.

Reply:
While this methodology has its benefit in integration tests, my primary concern about it is its maintainability as unit tests: is it easy to change? 

For example:

- what if we are adding another built-in schema
- what if we update `pretty_print` format
- what if we have some undetermined data in result set like random number, elapsed time, creation time..

Reply:
And note that this is not how data transfers in protocols, this is just how command line client displays data to users. It just represents part of all information in protocols, and is not the format programming language drivers consume data. As you mentioned, we missed type information with the assertions.

Reply:
I think the output format, or the `pretty_print` format, is seldom changed, because it's just MySQL cli's format, fixed. Also the format has no reason to change. 

Then it's the output's data that is subject to change, like some random numbers as you mentioned. In that case, testing like this is indeed not feasible. However, in that case, we can't build the expected output data as well (or building the expected output data is in excess of the efforts of not to, then people would switch the other way).

As for the output data's types, my point being that comparing to the benifit we got(that we gain much more understandability in testing cases), and the "erasure" of the types of the text protocol, we can tolerate not to verify it.

Reply:
Mark this as resolved so we can move on.